keyboards/handwired/wakizashi40/keymaps/default/keymap.c

```diff
@@ -0,0 +1,99 @@
+/* Copyright 2021 xiao (@xia0)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+[8:10 PM] я лов днкс: hey i was wondering if anyone could help me understand how to get a board onto VIA
+[8:11 PM] я лов днкс: from what I gather, i have to get it into the qmk repo then wait for the via team to update?
+[8:13 PM] sigprof: yes, first you need to make a PR to https://github.com/qmk/qmk_firmware with proper keyboard support, including the via keymap; then you make a PR with the VIA JSON to https://github.com/the-via/keyboards, providing a link to the QMK PR (there is a place for it in the template).
+[8:15 PM] я лов днкс: ty ty
+[8:17 PM] sigprof: also https://get.vial.today/docs/porting-to-via.html may be useful, even if you don't plan to use the Vial extensions
+*/
+
+#include QMK_KEYBOARD_H
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT_all(
+		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_BSPC,
+		MO(3), KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_ENT,
+		KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_ESC, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT,
+		KC_LCTL, KC_LGUI, KC_LALT, LT(1, KC_SPC), LT(2, KC_SPC), KC_LALT, KC_MENU, KC_LCTL),
+
+	[1] = LAYOUT_all(
+		KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F5, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12,
+		MO(11), KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_MINS, KC_TRNS, KC_EQL, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_SPC, KC_SPC, KC_TRNS, KC_TRNS, KC_TRNS),
+
+	[2] = LAYOUT_all(
```

Comment:
Please remove all these useless layers.

Reply:
Will this cause issues with having `#define DYNAMIC_KEYMAP_LAYER_COUNT 12` set in config.h? If I leave the extra layers in the via keymap it should work without issue?

Reply:
VIA should not be enabled on the default keymap, so the layer count matching the define is irrelevant here.